src/api/resources/base.py

```diff
@@ -720,3 +721,27 @@ def partial_update(self, request, pk=None):
 
     def destroy(self, request, pk=None):
         raise MethodNotAllowed("DELETE")
+
+
+class ArrayAggExt(ArrayAgg):
+    template = "ARRAY_REMOVE(%(function)s(%(distinct)s%(expressions)s %(ordering)s), NULL)"
```

Comment:
Just to confirm, I'm looking at the [source](https://github.com/django/django/blob/main/django/contrib/postgres/aggregates/general.py#L45) and wanting to make sure: currently if a fish species is not associated with any region, we get `[]`, right? This new approach will ensure we keep getting that, and not `[None]`, right? But the source code indicates that by passing `default=None` we could get what will be default in Django 5, i.e. `None`. Would the latter not work for us, thereby making the separate `regions_` source unnecessary? I get why we don't want `[None]`, but wouldn't `None` work?

Reply:
yes